src/frontend/src/handler/query.rs

```diff
@@ -45,10 +45,20 @@ pub async fn handle_query(context: OptimizerContext, stmt: Statement) -> Result<
     let query_mode = session.config().get_query_mode();
 
     debug!("query_mode:{:?}", query_mode);
+    let BoundStatement::Query(query) = &bound else {
+        unreachable!();
+    };
 
     let (data_stream, pg_descs) = match query_mode {
         QueryMode::Local => local_execute(context, bound)?,
-        QueryMode::Distributed => distribute_execute(context, bound).await?,
+        QueryMode::Distributed => {
+            if let BoundSetExpr::Select(select) = &query.body
+                && let Some(relation) = &select.from
+                && relation.contains_sys_table() {
+                return Err(internal_error("query on system table only works in local mode"));
+            }
+            distribute_execute(context, bound).await?
+        }
     };
```

Comment:
Sorry for late reply, but I don't think we should handle it like this here. When we found it contains sys table, we should just execute it in local mode rather than throwing an error. cc @yezizp2012

Reply:
Oh yes, we can force executing this query on local mode. 👍🏻 I will submit a new PR later.